File: Esame_22/JAVA/Esame.java

public class Esame {
	// Attributi
	protected int codice;
	protected String voto;

	// Costruttore
	public Esame(int codice, String voto) {
		this.codice = codice;
		this.voto = voto;
	}

	// Metodi
	// PUNTO 4: toString

	// conversione voto come intero per richiesta sul comando dell'esercizio 4
	public int getVotoAsInt() {
		if (this.voto.equals("30L"))
			return 31;
		else
			return Integer.parseInt(this.voto);
	}

	public String toString() {
		return "(" + codice + "," + voto + ")"; // guardare bene nell'output come fare
	}

	// PUNTO 5: metodo getter per la stampa nel punto 5
	public int getCorso() {
		return this.codice;
	}

	public String getVotoAsString() {
		return this.voto;
	}

} // fine classe

File: Esame_22/PYTHON/Esame.py
class Esame: # memorizza coppie codice voto

    # COSTRUTTORE
    def __init__(self, codice, voto):
        self._codice = codice
        self._voto = voto

    # METODI
    def getVotoAsInt(self):
        if  self._voto == "30L" :
            return 31
        else:
            return int(self._voto)

    def getVotoAsString(self):
        return self._voto

    def getCorso(self):
        return self._codice

    # per punto 4 degli studenti
    def toString(self):
        return "(" + str(self._codice) + "," + str(self._voto) + ")"
